packages/syntax-core/src/Badge/Badge.tsx

```diff
@@ -28,13 +30,18 @@ const textColorForBackgroundColor = (
  * [Badge](https://cambly-syntax.vercel.app/?path=/docs/components-badge--docs) is a component to display short text and give additional context to features and other components.
  */
 const Badge = ({
+  icon: Icon,
   text,
   color = "primary700",
 }: {
+  /**
+   * The icon to be displayed. Please use a [Material Icon](https://material.io/resources/icons/)
+   */
+  icon?: React.ComponentType<{ className: string }>;
   /**
    * The text to display inside the badge
    */
-  text: string;
+  text: string | ReactElement;
```

Comment:
```suggestion
  text: string;
```

Nope, not a ReactElement.  started off that direction, then went with `icon` prop after hearing opinions from @nathan-ong @somethiiing @jliotta